tests/tensorflow2/test_should_save_tensor.py

```diff
@@ -61,3 +61,9 @@ def test_should_save_tensor_with_custom_collection(out_dir):
         else:
             assert not hook.should_save_tensor_or_collection(layer_name, CollectionKeys.GRADIENTS)
             assert not hook.should_save_tensor_or_collection(layer_name, CollectionKeys.LAYERS)
+
+
+def test_should_save_tensor_behavior_without_hook_init(out_dir):
```

Comment:
nit: maybe reword to `test_should_save_tensor_behavior_before_collections_prepared`